### PR TITLE
Remove coach node

### PR DESCRIPTION
**Pull request by Ninjatest321**
This pull request adds a Referee PlayState subscription to all the files that use CoachState and don't have play state already (which is just Agent Action Client). This is part of the plan to remove CoachNode.

### CODE REVIEW

File: soccer/src/soccer/strategy/agent/agent_action_client.cpp [hunk 1]
```diff
@@ -95,6 +99,10 @@ void AgentActionClient::world_state_callback(const rj_msgs::msg::WorldState::Sha
     last_world_state_ = std::move(world_state);
 }
 
+void AgentActionClient::play_state_callback(const rj_msgs::msg::PlayState::SharedPtr& msg) {
+    return;
```
Comment (sid-parikh) [inline soccer/src/soccer/strategy/agent/agent_action_client.cpp]: Hey amogh, can you also change the AC & Position to get play state data from this topic? the callback is empty right now; we should be forwarding it to position. I know position doesn't use play state data right now, but that's okay

**Review by sid-parikh — CHANGES_REQUESTED** [review]

**Review by sid-parikh — APPROVED** [review]
Looks good, thanks!

**Comment by Ninjatest321** [comment]
This last commit calculates global overrides in the Planner Node (planner_node.cpp)

**Comment by p-nayak11** [comment]
Look at PR #2131

**Pull request closed without merging**